tests/Functional/Schema/MySQLSchemaManagerTest.php

```diff
@@ -544,6 +544,44 @@ public function testEnsureTableOptionsAreReflectedInMetadata(): void
         ], $onlineTable->getOption('create_options'));
     }
 
+    public function testCreatedCompositeForeignKeySchemaDoesNotDifferAfterCreation(): void
+    {
+        $this->connection->executeStatement('DROP TABLE IF EXISTS test_table_metadata');
+
+        $sql = <<<'SQL'
+CREATE TABLE test_table_foreign_key_referenced (
+  col1 INT NOT NULL,
+  col2 INT NOT NULL,
+  PRIMARY KEY (col2, col1)
+)
+COLLATE utf8_general_ci
+ENGINE InnoDB
+ROW_FORMAT COMPRESSED
+COMMENT 'Test foreign key referenced table'
+SQL;
+
+        $this->connection->executeStatement($sql);
+
+        $table = new Table('test_table_with_foreign_key');
+
+        $table->addOption('collate', 'utf8_general_ci');
+        $table->addOption('charset', 'utf8');
+        $table->addOption('comment', 'Test table with foreign key');
+
+        $table->addColumn('id', Types::INTEGER, ['autoincrement' => true, 'notNull' => true]);
+        $table->addColumn('child_col1', Types::INTEGER, ['notNull' => true]);
+        $table->addColumn('child_col2', Types::INTEGER, ['notNull' => true]);
+        $table->setPrimaryKey(['id']);
+
+        $table->addForeignKeyConstraint('test_table_foreign_key_referenced', ['child_col2', 'child_col1'], ['col2', 'col1']);
+        $this->schemaManager->createTable($table);
+
+        $tableFetched = $this->schemaManager->listTableDetails('test_table_with_foreign_key');
+
+        $diff = (new Comparator())->diffTable($tableFetched, $table);
```

Comment:
Since the issue is not about the comparator, it's redundant in the test as well. It's sufficient to get the foreign key definition of the introspected table and assert that the columns are in the right order.

Reply:
@morozov Makes sense.